quiche/src/lib.rs

```diff
@@ -18940,93 +18977,145 @@ mod tests {
             .set_application_protos(&[b"proto1", b"proto2"])
             .unwrap();
         config.verify_peer(false);
-        config.set_initial_max_data(100000);
-        config.set_initial_max_stream_data_bidi_local(100000);
-        config.set_initial_max_stream_data_bidi_remote(100000);
-        config.set_initial_max_streams_bidi(2);
-        config.set_active_connection_id_limit(4);
-        config.set_max_send_udp_payload_size(1350);
-        config.set_max_recv_udp_payload_size(1350);
+        config.set_max_send_udp_payload_size(1400);
         config.discover_pmtu(true);
 
-        // Perform initial handshake.
         let mut pipe = testing::Pipe::with_config(&mut config).unwrap();
         assert_eq!(pipe.handshake(), Ok(()));
 
-        let server_addr = testing::Pipe::server_addr();
-        let client_addr = testing::Pipe::client_addr();
-        let pid_1 = pipe
-            .server
-            .paths
-            .path_id_from_addrs(&(server_addr, client_addr))
-            .expect("no such path");
-
-        // Check that PMTU params are configured correctly
-        let pmtu_param = &mut pipe.server.paths.get_mut(pid_1).unwrap().pmtud;
-        assert!(pmtu_param.get_should_probe());
-        assert_eq!(pmtu_param.get_probe_size(), 1350);
-        assert_eq!(pipe.advance(), Ok(()));
+        // Verify PMTUD is enabled and ready to probe
+        let active_path = pipe.client.paths.get_active_mut().unwrap();
+        assert!(active_path.pmtud.is_enabled());
+        assert!(active_path.pmtud.get_should_probe());
+        let initial_probe_size = active_path.pmtud.get_probe_size();
+        assert_eq!(initial_probe_size, 1400);
 
-        for (_, p) in pipe.server.paths.iter_mut() {
-            assert_eq!(p.pmtud.get_current_mtu(), 1350);
-            assert!(!p.pmtud.get_should_probe());
+        let mut frames: Vec<frame::Frame> = Vec::new();
+        for _ in 0..2 {
+            let mut buf = [0; 1400];
+            let (len, _) = pipe.client.send(&mut buf).unwrap();
+            frames.append(
+                testing::decode_pkt(&mut pipe.server, &mut buf[..len])
+                    .unwrap()
+                    .as_mut(),
+            );
         }
+
+        assert!(matches!(frames[0], frame::Frame::ACK { .. }));
+        assert!(matches!(frames[1], frame::Frame::Padding { .. }));
+
+        let mut buf = [0; 1400];
+        assert_eq!(pipe.client.send(&mut buf).unwrap_err(), Error::Done);
+
+        // Verify probe flag was reset after sending
+        let active_path = pipe.client.paths.get_active_mut().unwrap();
+        assert!(!active_path.pmtud.get_should_probe());
     }
 
     #[rstest]
-    fn pmtud_probe_loss(
+    /// Test that probe flag is properly re-enabled after probe loss
+    fn pmtud_probe_flag_reenabled_after_loss(
```

Comment:
This test seems redundant with `pmtud_probe_retry_after_loss` further down.

Reply:
Indeed, I removed it